cabal install hmatrix-tests fails on 8.8.1 and 8.10.1 alpha

[3 of 4] Compiling Numeric.LinearAlgebra.Tests ( src/Numeric/LinearAlgebra/Tests.hs, dist/build/Numeric/LinearAlgebra/Tests.o, dist/build/Numeric/LinearAlgebra/Tests.dyn_o )

src/Numeric/LinearAlgebra/Tests.hs:410:5: error:
    ‘fail’ is not a (visible) method of class ‘Monad’
    |
410 |     fail _   = MaybeT $ return Nothing
    |     ^^^^
cabal: Failed to build hmatrix-tests-0.19.0.0. See the build log above for
details.

--- a/packages/tests/hmatrix-tests.cabal
+++ b/packages/tests/hmatrix-tests.cabal
@@ -25,11 +25,13 @@ flag gsl
   default: True
 
 library
-
-    Build-Depends:      base >= 4 && < 5, deepseq,
-                        QuickCheck >= 2, HUnit, random,
-                        hmatrix >= 0.18
+    Build-Depends:      base >= 4 && < 5
                       , binary
+                      , deepseq
+                      , hmatrix >= 0.18
+                      , HUnit
+                      , random
+                      , QuickCheck >= 2
     if flag(gsl)
       Build-Depends:    hmatrix-gsl >= 0.18
 

--- a/packages/tests/src/Numeric/LinearAlgebra/Tests.hs
+++ b/packages/tests/src/Numeric/LinearAlgebra/Tests.hs
@@ -430,11 +430,8 @@ instance Monad m => Monad (MaybeT m) where
                         case res of
                                  Nothing -> return Nothing
                                  Just r  -> runMaybeT (f r)
-    fail _   = MaybeT $ return Nothing
 
-lift_maybe m = MaybeT $ do
-                        res <- m
-                        return $ Just res
+lift_maybe = MaybeT . fmap Just
 
 -- apply a test to successive elements of a vector, evaluates to true iff test passes for all pairs
 --successive_ :: Storable a => (a -> a -> Bool) -> Vector a -> Bool
@@ -443,7 +440,7 @@ successive_ t v = maybe False (\_ -> True) $ evalState (runMaybeT (mapVectorM_ s
                   ep <- lift_maybe $ state_get
                   if t e ep
                      then lift_maybe $ state_put e
-                     else (fail "successive_ test failed")
+                     else MaybeT $ return Nothing
 
 -- operate on successive elements of a vector and return the resulting vector, whose length 1 less than that of the input
 --successive :: (Storable a, Storable b) => (a -> a -> b) -> Vector a -> Vector b

